### PR TITLE
[PM-4871] Remove clickable row in favor of clickable cell content

### PR DESCRIPTION
**Pull request by Spitfireap**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ X] UX Improvement
```

## Objective

Improve UX.

When selecting multiple items in the vault it is easy to mis-click and trigger the "show item" action which not only show the item information but also uncheck every previously checked checkbox. It is really hard to manage a great number of items.

The proposed change check the click event and if the mouse is above the checkbox's `<td>` the checkbox will trigger instead. 

[Forum link referring to this issue.](https://community.bitwarden.com/t/please-remove-clickable-area-around-checkbox/54417)

## Code changes

- **vault-cipher-row.component.html:** added an id to the `<td>` containing the checkbox for the click event check
- **vault-cipher-row.component.ts:** Updated the click event by providing the event's target. If the target contains the id, the input is then checked.

## Screenshots

old behavior: 

https://github.com/bitwarden/clients/assets/45575529/2b5c4979-4b4e-4ee4-910c-92489dd5d695

new behavior:

https://github.com/bitwarden/clients/assets/45575529/cc804d9e-4ac8-4f53-a825-4539a6e6b673

### CODE REVIEW

**Review by danielleflinn — APPROVED** [review]
This behavior seems to align with the Design request outlined in: [PM-3078](https://bitwarden.atlassian.net/browse/PM-3078)

@gbubemismith when you get to reviewing this for Vault please reference the internal task which outlines our desired fix for this UX bug.

File: apps/web/src/app/vault/components/vault-items/vault-cipher-row.component.ts [hunk 1]
```diff
@@ -52,12 +52,20 @@ export class VaultCipherRowComponent {
...
   }
 
-  @HostListener("click")
-  protected click() {
-    this.router.navigate([], {
-      queryParams: { itemId: this.cipher.id },
-      queryParamsHandling: "merge",
-    });
+  @HostListener("click", ["$event.target"])
+  protected click(target: HTMLSpanElement) {
```
Comment (gbubemismith) [inline apps/web/src/app/vault/components/vault-items/vault-cipher-row.component.ts:56]: nitpick: I think the target parameter can be changed to `HTMLElement` as it is a more general type.

**Review by gbubemismith — CHANGES_REQUESTED** [review]
@Spitfireap Thanks for fixing this bug. I have one suggestion, and we can get this approved.

File: apps/web/src/app/vault/components/vault-items/vault-cipher-row.component.ts (hunk 1 above)
Comment (Spitfireap) [reply apps/web/src/app/vault/components/vault-items/vault-cipher-row.component.ts]: @gbubemismith Change committed ;). 
The lint process applied some change on L.37. Do you want to keep them ?

Comment (Spitfireap) [reply apps/web/src/app/vault/components/vault-items/vault-cipher-row.component.ts]: My pleasure and done ! Thanks for your review.
I've deleted L.37 altogether (`activatedRouter` didn't seem to be used before). I'm not fluent enough in Angular to be sure it wasn't needed (even though I didn't see any regression). I would appreciate it if you could double-check.
On a side note I feel that making the clickable area for the checkbox wider than its visual appearance (as proposed initially) would still be an accessibility improvement...

Comment (gbubemismith) [reply apps/web/src/app/vault/components/vault-items/vault-cipher-row.component.ts]: One last thing: can you fix the conflict on your branch?

**Review by gbubemismith — APPROVED** [review]

**Comment by gbubemismith** [comment]
Hello @Spitfireap, Firstly, thank you for your work on this bug. I moved this to our QA team to test and received feedback on adding this behaviour to `collections` to keep things consistent. Given the excellent work you have already done, would you mind including this adjustment on the `VaultCollectionRowComponent`? Please know there's no pressure to take this on if you're not comfortable.

**Comment by gbubemismith** [comment]
> @gbubemismith done :). So no discussion possible about making the clickable area for the checkbox bigger ?

Thanks for your quick response. I have started this discussion and am sure this suggestion will be considered.

**Review by gbubemismith — APPROVED** [review]

**Comment by gbubemismith** [comment]
@Spitfireap Our QA team has approved this PR 🚀. Thanks for your work on this.

**Pull request merged**